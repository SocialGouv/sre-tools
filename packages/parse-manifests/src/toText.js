import parseManifests from "./index.js";

import {
  getGrafanaLogsUrl,
  getGrafanaWorkloadsUrl,
  getRancherUrls,
  getGrafanaPodsUrl,
} from "./tools.js";

const toText = (manifests) => {
  const parsed = parseManifests(manifests);

  return `
### Ingresses

${parsed.hosts.map((host) => ` - 🚀 https://${host}`).join("\n")}
${
  (parsed.redirects.length &&
    `
### Redirects

${parsed.redirects
  .map(({ from, to }) => ` - https://${from} ➡️ ${to}`)
  .join("\n")}
`) ||
  ""
}
### Docker images

${parsed.images.map((image) => ` - 📦 docker pull ${image}`).join("\n")}

### Debug

#### 📕 Loki logs for namespace ${parsed.namespace}:

${getGrafanaLogsUrl(parsed)}

#### 📈 Pods monitoring for namespace ${parsed.namespace}:

${getGrafanaPodsUrl(parsed)}

#### 📈 Workloads monitoring for namespace ${parsed.namespace}:

${getGrafanaWorkloadsUrl(parsed)}

#### 👮‍♂️ Rancher project ${parsed.namespace}:

${getRancherUrls(parsed)
  .map(({ name, url }) => ` - ${name} : ${url}`)
  .join("\n")}

`;
};

export default toText;
