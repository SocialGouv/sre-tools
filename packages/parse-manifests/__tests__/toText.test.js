import toText from "../src/toText.js";

import fs from "fs";
import path from "path";

//console.log(parseManifests(fs.readFileSync("../sample.yml").toString()));

test("should render text", () => {
  const yaml = fs.readFileSync(path.join("sample.yml")).toString();
  const redirect = fs.readFileSync(path.join("redirect.yml")).toString();
  expect(toText(yaml + redirect)).toMatchSnapshot();
});

test("should render text without redirect", () => {
  const yaml = fs.readFileSync(path.join("sample.yml")).toString();
  expect(toText(yaml)).toMatchSnapshot();
});
