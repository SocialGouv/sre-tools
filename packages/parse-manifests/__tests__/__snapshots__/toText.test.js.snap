// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render text 1`] = `
"
### Ingresses

 - 🚀 https://backend-app-preprod.dev.company.com/
 - 🚀 https://version1-app-preprod.dev.company.com/
 - 🚀 https://version1-app-preprod.dev.company.com/something
 - 🚀 https://version2-app-preprod.dev.company.com/

### Redirects

 - https://www.some.url ➡️ https://target.fr$request_uri
 - https://www.some.alternative.url ➡️ https://target.fr$request_uri

### Docker images

 - 📦 docker pull ghcr.io/company/app/backend:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version1:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version2:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/docker/wait-for-postgres:6.38.3

### Debug

#### 📕 Loki logs for namespace app-preprod:

https://grafana.fabrique.social.gouv.fr/explore?orgId=1&left=%5B%22now-6h%22,%22now%22,%22Loki%22,%7B%22expr%22:%22%7Bnamespace%3D%5C%22app-preprod%5C%22%7D%22%7D%5D

#### 📈 Pods monitoring for namespace app-preprod:

https://grafana.fabrique.social.gouv.fr/d/85a562078cdf77779eaa1add43ccec1e/kubernetes-compute-resources-namespace-pods?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod

#### 📈 Workloads monitoring for namespace app-preprod:

https://grafana.fabrique.social.gouv.fr/d/a87fb0d919ec0ea5f6543124e16c42a5/kubernetes-compute-resources-namespace-workloads?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod&var-type=deployment

 - CNPG some-cnpg-cluster: https://grafana.fabrique.social.gouv.fr/d/z7FCA4Nn1/cloudnativepg?orgId=1&refresh=30s&var-DS_PROMETHEUS=default&var-namespace=app-preprod&var-cluster=some-cnpg-cluster&var-instances=All

"
`;

exports[`should render text with rancher details 1`] = `
"
### Ingresses

 - 🚀 https://backend-app-preprod.dev.company.com/
 - 🚀 https://version1-app-preprod.dev.company.com/
 - 🚀 https://version1-app-preprod.dev.company.com/something
 - 🚀 https://version2-app-preprod.dev.company.com/

### Redirects

 - https://www.some.url ➡️ https://target.fr$request_uri
 - https://www.some.alternative.url ➡️ https://target.fr$request_uri

### Docker images

 - 📦 docker pull ghcr.io/company/app/backend:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version1:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version2:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/docker/wait-for-postgres:6.38.3

### Debug

#### 📕 Loki logs for namespace app-preprod:

https://grafana.fabrique.social.gouv.fr/explore?orgId=1&left=%5B%22now-6h%22,%22now%22,%22Loki%22,%7B%22expr%22:%22%7Bnamespace%3D%5C%22app-preprod%5C%22%7D%22%7D%5D

#### 📈 Pods monitoring for namespace app-preprod:

https://grafana.fabrique.social.gouv.fr/d/85a562078cdf77779eaa1add43ccec1e/kubernetes-compute-resources-namespace-pods?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod

#### 📈 Workloads monitoring for namespace app-preprod:

https://grafana.fabrique.social.gouv.fr/d/a87fb0d919ec0ea5f6543124e16c42a5/kubernetes-compute-resources-namespace-workloads?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod&var-type=deployment

 - CNPG some-cnpg-cluster: https://grafana.fabrique.social.gouv.fr/d/z7FCA4Nn1/cloudnativepg?orgId=1&refresh=30s&var-DS_PROMETHEUS=default&var-namespace=app-preprod&var-cluster=some-cnpg-cluster&var-instances=All

"
`;

exports[`should render text without ingress 1`] = `
"


### Docker images

 - 📦 docker pull ghcr.io/company/app/backend:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version1:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version2:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/docker/wait-for-postgres:6.38.3

### Debug

#### 📕 Loki logs for namespace app-preprod:

https://grafana.fabrique.social.gouv.fr/explore?orgId=1&left=%5B%22now-6h%22,%22now%22,%22Loki%22,%7B%22expr%22:%22%7Bnamespace%3D%5C%22app-preprod%5C%22%7D%22%7D%5D

#### 📈 Pods monitoring for namespace app-preprod:

https://grafana.fabrique.social.gouv.fr/d/85a562078cdf77779eaa1add43ccec1e/kubernetes-compute-resources-namespace-pods?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod

#### 📈 Workloads monitoring for namespace app-preprod:

https://grafana.fabrique.social.gouv.fr/d/a87fb0d919ec0ea5f6543124e16c42a5/kubernetes-compute-resources-namespace-workloads?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod&var-type=deployment



"
`;

exports[`should render text without redirect 1`] = `
"
### Ingresses

 - 🚀 https://backend-app-preprod.dev.company.com/
 - 🚀 https://version1-app-preprod.dev.company.com/
 - 🚀 https://version1-app-preprod.dev.company.com/something
 - 🚀 https://version2-app-preprod.dev.company.com/

### Docker images

 - 📦 docker pull ghcr.io/company/app/backend:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version1:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/app/frontend-version2:1.1.0-alpha.6
 - 📦 docker pull ghcr.io/company/docker/wait-for-postgres:6.38.3

### Debug

#### 📕 Loki logs for namespace app-preprod:

https://grafana.fabrique.social.gouv.fr/explore?orgId=1&left=%5B%22now-6h%22,%22now%22,%22Loki%22,%7B%22expr%22:%22%7Bnamespace%3D%5C%22app-preprod%5C%22%7D%22%7D%5D

#### 📈 Pods monitoring for namespace app-preprod:

https://grafana.fabrique.social.gouv.fr/d/85a562078cdf77779eaa1add43ccec1e/kubernetes-compute-resources-namespace-pods?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod

#### 📈 Workloads monitoring for namespace app-preprod:

https://grafana.fabrique.social.gouv.fr/d/a87fb0d919ec0ea5f6543124e16c42a5/kubernetes-compute-resources-namespace-workloads?orgId=1&refresh=10s&var-datasource=default&var-cluster=dev2&var-namespace=app-preprod&var-type=deployment

 - CNPG some-cnpg-cluster: https://grafana.fabrique.social.gouv.fr/d/z7FCA4Nn1/cloudnativepg?orgId=1&refresh=30s&var-DS_PROMETHEUS=default&var-namespace=app-preprod&var-cluster=some-cnpg-cluster&var-instances=All

"
`;
